### PR TITLE
Modernize/refactor formula

### PR DESCRIPTION
**Pull request by alxwr**
- split formula into reusable parts/modules/states
- updated repo paths
- use YAML files to make `map.jinja` more readable
- fixed `autoconfig.php`
- added a cron job to trigger background tasks
- tested on Ubuntu 16.04 and FreeBSD 10.3

### CODE REVIEW

File: owncloud/map.jinja [hunk 1]
```diff
@@ -1,39 +1,36 @@
...
+
+{% set owncloud = salt['grains.filter_by'](
+    {'default': default_settings}
+, merge=salt['grains.filter_by'](
+    osfamilymap,
+    grain='os_family'
+, merge=salt['grains.filter_by'](
+    {
+        'default': {
+            'client_pkg_repo': "http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/{}_{}/ /".format(grains['os'], grains['osrelease']),
```
Comment (whiteinge) [inline owncloud/map.jinja]: We should have Python 2.6 support for a tad longer, so we need indexed placeholders here.

Comment (alxwr) [reply owncloud/map.jinja]: Thanks! I'll fix that as soon as I find some time to do so.

**Comment by alxwr** [comment]
@whiteinge @aboe76  Added indexed placeholders. (Longest wait ever for such a trivial patch. Sry.)

**Comment by whiteinge** [comment]
Thanks for the bump. Sorry I missed the updates and for that long wait!

**Comment by alxwr** [comment]
@whiteinge No, I meant, **I** was taking too long. No need to apologize! :-)
Thanks for the merge!

**Pull request merged**